pyfr/writers/nodemaps.py

```diff
@@ -0,0 +1,386 @@
+        ('pri', 75): np.array([0,  4, 14, 60, 64,
+                               74,  1,  2,  3,  8,
+                               11, 13, 12,  9,  5,
+                               61, 62, 63, 68, 71,
+                               73, 72, 69, 65, 15,
+                               30, 45, 19, 34, 49,
+                               29, 44, 59,  6,  7,
+                               10, 66, 67, 70, 16,
+                               17, 18, 31, 32, 33,
+                               46, 47, 48, 23, 26,
+                               28, 38, 41, 43, 53,
+                               56, 58, 20, 24, 27,
+                               35, 39, 42, 50, 54,
+                               57, 21, 22, 25, 36,
+                               37, 40, 51, 52, 55]),
+        ('pri', 126): np.array([0,   5,  20, 105,
+                                110, 125,   1,   2,
+                                3,   4,  10,  14,
+                                17,  19,  18,  15,
+                                11,   6, 106, 107,
+                                108, 109, 115, 119,
+                                122, 124, 123, 120,
+                                116, 111,  21,  42,
+                                63,  84,  26,  47,
+                                68,  89,  41,  62,
+                                83, 104,   7,   8,
+                                9,  12,  13,  16,
+                                112, 113, 114, 117,
+                                118, 121,  22,  23,
+                                24,  25,  43,  44,
+                                45,  46,  64,  65,
+                                66,  67,  85,  86,
+                                87,  88,  31,  35,
+                                38,  40,  52,  56,
+                                59,  61,  73,  77,
+                                80,  82,  94,  98,
+                                101, 103,  27,  32,
+                                36,  39,  48,  53,
+                                57,  60,  69,  74,
+                                78,  81,  90,  95,
+                                99, 102,  28,  29,
+                                30,  33,  34,  37,
+                                49,  50,  51,  54,
+                                55,  58,  70,  71,
+                                72,  75,  76,  79,
+                                91,  92,  93,  96,
+                                97, 100]),
+    }
+
+    to_pyfr = {k: np.argsort(v) for k, v in from_pyfr.items()}
+
+
+class VTK9NodeMaps(object):
+    """
+    WARNING
+    VTK9 changed the connectivity mapping of vtkLagrangeHexahedron
+    cells. However, if the `xml version` token of a given `vtu` or
+    `pvtu` file is lower than 5.1, then VTK>=9 will suppose that the
+    connectivity is consistent with that of VTK8 and will reorder the
+    nodes without the user noticing it.
+    As `xml version` is hardcoded to 1.0 in PyFR, we shouldn't use
```

Comment:
Any reason we don't change the version we write out?

Reply:
I didn't want to introduce new regressions with this PR. Moreover, XML VTK files are usually poorly documented (see https://gitlab.kitware.com/vtk/vtk/-/issues/17675 or https://gitlab.kitware.com/vtk/vtk/-/issues/17989) so I do not know what may go wrong if one changes the value of `xml version` ... Anyway, I'll have a look at it soon and try to see if there are any regressions with different `xml version`

Reply:
I think the XML version should always be 1 as that is an XML standard thing.  It is the version="0.1" thing which might need to change.  If we want to be careful we can make the version a property of the class and have the high-order subclass bump it up.

Reply:
Yes sorry I was referring to that token my bad. So it seems that for the `XML` like files (`vtu` for example) the switch is done for `version < 2.1` and for `Legacy` type files the switch is done for `version < 5.1`. I have tried using `version = "2.1"` and only the VTK9 connectivity gives correct results. So I'll add a property to each `vtk` writer with an appropriate `version` (0.1 for `vtk.py` and 2.1 for `vtkho.py` and I'll use VTK9 connectivity when needed (do you want to leave VTK8 connectivity within the files?)

Reply:
I think we can get rid of the VTK8 connectivity.